src/write.c

```diff
@@ -4,16 +4,26 @@
 #include "avif/internal.h"
 
 #include <assert.h>
+#include <stdint.h>
 #include <string.h>
 #include <time.h>
 
-#define MAX_ASSOCIATIONS 16
-struct ipmaArray
+// Section 8.11.14.2 of ISO/IEC 14496-12 (ItemPropertyAssociationBox 'ipma' syntax):
+//   if (flags & 1)
+//     unsigned int(15) property_index;
+//   else
+//     unsigned int(7) property_index;
+//
+// libavif writes 'ipma' with flags set to 0.
+#define MAX_PROPERTY_INDEX (1 << 7) - 1
```

Comment:
Parenthesize the value of the macro:

```
#define MAX_PROPERTY_INDEX ((1 << 7) - 1)
```

It is a good practice to parenthesize the value of a macro and to parenthesize macro parameters (if any) in the value of a macro.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/2560